lib/faktory/configuration.ex

```diff
@@ -10,216 +10,230 @@ defmodule Faktory.Configuration do
+  worker side with sane defaults. You can call `Faktory.Configuration.all/0`
+  to see these defaults.
+
+  ### Client Configuration
+
+    Settings used for _enqueuing_ jobs.
 
-  Notice that client/worker specific options will inherit from top level options
-  where applicable. In the above example, `client.host` will be `"localhost"`.
+    ```elixir
+    config :faktory_worker_ex, FooConfig,
+      adapter: Faktory.Configuration.Client
+      host: "foo_faktory.mycompany.com",
 
-  ### Options
+    config :faktory_worker_ex, BarConfig
+      adapter: Faktory.Configuration.Client
+      host: "bar_faktory.mycompany.com"
+    ```
 
-  * `host` - Faktory server host. Default `"localhost"`
-  * `port` - Faktory server port. Default `7419`
-  * `password` - Faktory server password. Default `nil`
-  * `config_fn` - Callback function for runtime config. Default `nil`
+    Valid options:
 
-  ### Client Options
+    * `host` - Host of Faktory server. Default `"localhost"`
+    * `port` - Port of Faktory server. Default `7419`
+    * `pool` - Connection pool size. Default `10`
+    * `middleware` - Middleware chain. Default `[]`
+    * `password` - For Faktory server authentication. Default `nil`
+    * `use_tls` - Connect to Faktory server using TLS. Default `false`
 
-    * `pool` - Client connection pool size. Default `10`
-    * `middleware` - Client middleware chain. Default `[]`
+  ### Default Client
+
+  The first configured client is the "default client" and is used by
+  `Faktory.Job.perform_async/2` when no client is specified.
+
+  ```elixir
+  # No client is specified, default client is used.
+  MySuperJob.perform_async([1, 2, 3])
+
+  # Explicitly specify a client.
+  MyUltraJob.perform_async([1, 2, 3], client: BarConfig)
+  ```
 
   ### Worker Options
 
-    * `pool` - Worker connection pool size. Default `${concurrency}`
-    * `middleware` - Worker middleware chain. Default `[]`
-    * `concurrency` - How many worker processes to start. Default `20`
+    ```elixir
+    config :faktory_worker_ex, FooWorker,
+      adapter: Faktory.Configuration.Worker,
+      host: "foo_faktory.mycompany.com"
+
+    config :faktory_worker_ex, BarWorker,
+      adapter: Faktory.Configuration.Worker,
+      host: "bar_faktory.mycompany.com"
+    ```
+
+    Valid options:
+
+    * `host` - Host of Faktory server. Default `"localhost"`
+    * `port` - Port of Faktory server. Default `7419`
+    * `concurrency` - How many concurrent jobs to process. Default `20`
+    * `pool` - Connection pool size. Default `${concurrency}`
+    * `middleware` - Middleware chain. Default `[]`
     * `queues` - List of queues to fetch from. Default `["default"]`
 
   ### Runtime Configuration
 
-  You can specify a callback to do runtime configuration. For example, to read
-  host and port from environment variables.
+  There are two ways to do runtime configuration:
+  1. A special syntax to read environment vars
```

Comment:
IMO, it would be clearer to say "The conventional `{:system, "MY_VAR"}` syntax" (or something similar) since that syntax is fairly common among Elixir libs.